src/frontend/js/utils/api/lms/openedx-hawthorn.ts

```diff
@@ -20,8 +20,10 @@ const API = (
   APIConf: AuthenticationBackend | LMSBackend,
   options?: ApiOptions,
 ): ApiImplementation => {
-  const extractCourseIdFromUrl = (url: string): Maybe<Nullable<string>> =>
-    url.match((APIConf as LMSBackend).course_regexp)?.groups?.course_id;
+  const extractCourseIdFromUrl = (url: string): Maybe<Nullable<string>> => {
+    const matches = url.match((APIConf as LMSBackend).course_regexp);
+    return matches && matches[1] ? matches[1] : null;
```

Comment:
I'm wondering, do you see anyway to lint that this is not used in future code?

Reply:
For this case, as the regex is provided by the backend it is something hard to detect from the frontend.

Reply:
ok... You could look for `.groups` but that may be a bit too common...

Reply:
The error occured when the regex is instantiate. We couldn't even test whether the `groups` attribute exists. We could try to detect if the regex course string contains `<?.*>` but from my point of view it is not a good idea to add this complexity to support old browsers.

Reply:
:+1: I follow you.